components/script/script_thread.rs

```diff
@@ -1836,6 +1840,23 @@ impl ScriptThread {
 
         self.documents.borrow_mut().insert(incomplete.pipeline_id, &*document);
 
+        // Notify iframe that the new document is active
+        if let Some(frame_element) = frame_element {
+            // This is a load occuring inside and iframe in this script thread.
+            // The iframe must update its current PipelineId.
+            frame_element.update_pipeline_id(incomplete.pipeline_id, false);
+        } else if incomplete.parent_info.is_some() {
+            // This is a load occuring inside an iframe in another script thread.
+            // The iframe must be synchronously notified to update its current PipelineId.
```

Comment:
IRC: http://logs.glob.uno/?c=mozilla%23servo&s=11+Jan+2017&e=11+Jan+2017#c591565 (around there).
The `webstorage` tests revealed an edge case where we have to synchronously update the `PipelineId` of the iframe before the new document executes anything. Otherwise the iframe contentDocument's URL wont be updated to match the URL from the webstorage event